src/main/java/io/confluent/connect/hdfs/wal/WALFile.java

```diff
@@ -308,17 +312,27 @@ private void writeFileHeader()
 
     @Override
     public synchronized void close() throws IOException {
-      keySerializer.close();
-      valSerializer.close();
-      if (out != null) {
-        // Close the underlying stream iff we own it...
-        if (ownOutputStream) {
-          out.close();
-          fs.close();
-        } else {
-          out.flush();
+      try {
+        keySerializer.close();
+        valSerializer.close();
+        if (out != null) {
+          // Close the underlying stream iff we own it...
+          if (ownOutputStream) {
+            out.close();
+          } else {
+            out.flush();
+          }
+          out = null;
```

Comment:
Should this be in the try-finally block? Right now, if `out.close()` or `out.flush()` throws an exception, we'll not null `out` and, if `close()` is called more than once (albeit unnecessarily) then we'd call close or flush another time. It's probably simpler to just ensure we null `out` in the finally block.